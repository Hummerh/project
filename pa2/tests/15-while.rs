fn main() {
      while (false && true) {
            while (true || true) {
                  while (1 < 2) {
                        1;
                        2;
                        3;
                  };
                  4;
            };
            5;
      };
      6;
}
