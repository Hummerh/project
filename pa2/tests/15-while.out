(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (while
                    (and
                        (false)
                        (true))
                    (block
                        (while
                            (or
                                (true)
                                (true))
                            (block
                                (while
                                    (lt
                                        (lit-dec)
                                        (lit-dec))
                                    (block
                                        (lit-dec)
                                        (lit-dec)
                                        (lit-dec)))
                                (lit-dec)))
                        (lit-dec)))
                (lit-dec)))))